sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,183 @@
+
+        if (key_vault_options is not None and
+                (key_vault_options.credential is not None or key_vault_options.secret_clients is not None or
+                 key_vault_options.secret_resolver is not None)):
+            correlation_context += ",UsesKeyVault"
+
+        headers["Correlation-Context"] = correlation_context
+        useragent = USER_AGENT
+
+        if connection_string is not None:
+            self._client = AzureAppConfigurationClient.from_connection_string(
+                connection_string, user_agent=useragent, headers=headers)
+            return
+        self._client = AzureAppConfigurationClient(
+            endpoint, credential, user_agent=useragent, headers=headers)
+
+    def __resolve_keyvault_references(self, config, key_vault_options, secret_clients):
+        if key_vault_options is None:
+            raise AttributeError(
+                "Key Vault options must be set to resolve Key Vault references.")
+        j_object = json.loads(config.value)
+        uri_value = j_object['uri']
+
+        uri = urlparse(uri_value)
+
+        key_vault_uri = "https://" + uri.hostname
+        key_vault_secret_prefix = "/secrets/"
+        key_vault_secret_name = uri.path[len(
+            key_vault_secret_prefix):]
+        if key_vault_options.credential is not None:
+            secret_client = None
+
+            # Clients only should be made once, will reuse if client already made
+            for client_uri in secret_clients:
+                if client_uri == key_vault_uri:
+                    secret_client = secret_clients[client_uri]
+                    break
+            if secret_client is None:
+                secret_client = SecretClient(
+                    vault_url=key_vault_uri, credential=key_vault_options.credential)
+                secret_clients[key_vault_uri] = secret_client
+                secret = secret_client.get_secret(
+                    key_vault_secret_name)
+                self._dict[self.__trim(config.key)] = secret.value
+                return
+            raise AttributeError(
+                "No Secret Client found for Key Vault reference %s%s" % (key_vault_uri, uri.path))
+        if key_vault_options.secret_clients is not None:
+            for secret_client in key_vault_options.secret_clients:
+                if secret_client.vault_url == key_vault_uri:
+                    secret = secret_client.get_secret(
+                        key_vault_secret_name)
+                    self._dict[self.__trim(
+                        config.key)] = secret.value
+                    return
+        if key_vault_options.secret_resolver is not None:
+            self._dict[self.__trim(
+                config.key)] = key_vault_options.secret_resolver(uri)
+            return
+        raise AttributeError(
+            "No Secret Client found for Key Vault reference %s%s" % (key_vault_uri, uri.path))
+
+    def __trim(self, key):
+        for trim in self._trim_prefixes:
```

Comment:
Since multiple prefixes could start with the same characters, we need to trim the longest prefix first. To achieve this, we can sort the user provided prefixes in descending order.

Reply:
I now sort them before adding them to _trim_prefixes.